### ISSUE TITLE
Which SoC for RFID Reader? Raspberry Pi Zero W ?

### ISSUE THREAD

**Issue by nelsonic** [nelsonic/door#2]
The idea is to use a Raspberry Pi (_or Arduino_) to power the RFID Reader for the Door(s).

The Raspberry Pi Zero W has the following _advantages_:
+ Cheaper (_**£9.16** per board_) ... see screenshot below.
+ Lower power consumption (_can be "fed" from the 2.1A USB Wall Outlet_)

https://thepihut.com/products/raspberry-pi-zero-w
![image](https://user-images.githubusercontent.com/194400/37708669-7a801c32-2cff-11e8-8cec-c3848a2911fb.png)

### Power Consumption Comparison

http://raspi.tv/2017/how-much-power-does-pi-zero-w-use
![image](https://user-images.githubusercontent.com/194400/37709809-4bb135b8-2d03-11e8-9d40-0e495dbcf7a6.png)

The "_official_" power consumption figures are a bit less informative:
https://www.raspberrypi.org/help/faqs/#powerReqs
![image](https://user-images.githubusercontent.com/194400/37710317-2bc98f6e-2d05-11e8-8497-f99c38a5a57e.png)

### RFID Reader Chip

https://www.aliexpress.com/item/RFID-RC522-Module-Card-Read-Antenna-RF-Reader-IC-Card-Proximity-Sensor-FZ0565/32397071273.html
![image](https://user-images.githubusercontent.com/194400/37710497-c0dc43da-2d05-11e8-9973-3f6f3560dfd5.png)

**Comment by nelsonic** [comment]
For the time being @th0mas is using the **`PN532`** https://github.com/dwyl/smart-home-security-system/issues/1#issuecomment-651036343
He has made great progress with it: https://github.com/dwyl/smart-home-security-system/blob/master/NFC_Reader.md


When we "deploy" the system, we will need a more robust (outdoor) solution.

Obviously I find an e-commerce site called "Bang Good" to be a bit dubious ... 🙄 
But this _appears_ to be the most reasonably priced IP65 Waterproof NFC/RFID Reader I can find:
https://www.banggood.com/Long-Range-RFID-Card-Reader-13_56MHZ125KHZ-Proximity-Card-Access-Control-Reader-Wiegand34-IP65-Waterproof-NFC-Reader-p-1534554.html
![image](https://user-images.githubusercontent.com/194400/87549441-b530d000-c6a5-11ea-9f8c-d8229479dc67.png)

### Spec:

Proximity 125KHz /13.56MHz  Card Reader
Read Range: Up to 10CM(125KHz)/5CM(13.56MHz)
34bit Wiegand code output
External LED Control
External Buzzer Control
Indoor / Outdoor Operation
Solid Epoxy Potted
Waterproof IP65
Reverse Polarity Protection

Read Range Up to 5 - 10CM
Reading Time (Card) <=300ms
Power / Current DC 6-14V / Max.70mA
Input Port 2ea (External LED Control, External Buzzer Control)
Output Format 34bit Wiegand (default)
LED Indicator 2 Color LED Indicators(Red and Green)
Beeper : Yes
Operating Temperature -20° to +65°C
Operating Humidity 10% to 90% relative humidity non-condensing
Color : Black
Material ABS+PC with texture
Dimension(W x H x T)mm : 116*75*17.3
Waterproof level : IP65
 
Package includeds :

1 x Long Range RFID Card Reader


![image](https://user-images.githubusercontent.com/194400/87549732-1658a380-c6a6-11ea-8644-70a528f69528.png)

it _appears_ to be possible to use this with a Raspberry Pi:
https://raspberrypi.stackexchange.com/questions/56706/raspberry-pi-as-wiegand-device-output-wiegand-to-receiver

**Comment by t0mhaines** [comment]
We would have to write our own NFC receiver library to use that with our System, as there doesn't seem to be any existing Elixir/(any?) libraries. My preference would be using a reader with the `PN532` chip in it as its so *common*, and we can lean on an existing implementation. 

A starter on the reading NFC chips using Wiegand: https://stackoverflow.com/questions/35491151/what-kind-of-data-can-a-wiegand-26-reader-read-from-an-nfc-card

**Comment by nelsonic** [comment]
@th0mas good point. 100% onboard with using a PN532-based reader that already has an `Elixir` library! ✅ 

We can revisit Wiegand in the future: https://www.aliexpress.com/item/4000459364534.html
![image](https://user-images.githubusercontent.com/194400/87551899-e3fc7580-c6a8-11ea-87d2-74df5de75a48.png)

**Comment by Uzzi** [comment]
I've https://www.amazon.it/gp/product/B07DK9R7K4/ref=ppx_yo_dt_b_asin_title_o00_s00?ie=UTF8&th=1

Can you share me exact level shifter model?
Can you share me pins connections scheme Raspberry---->level shifter model------>Reader?

Thank you

**Comment by nelsonic** [comment]
Wish we could help you @Uzzi ... sadly we don't have the answer.  😞 
Best of luck on your quest. 🔍 
If you find the answer, please share. 💭

**Comment by Uzzi** [comment]
Hi Could you help me about https://wiki.geekworm.com/Raspberry_Pi_DIDO_Board and https://ae01.alicdn.com/kf/H1af968b7b7ec4bce83c5e4152c6be2b2z.jpg connections schema diagram?
Thank you

**Comment by nelsonic** [comment]
@Uzzi wish we could help you but we are not the experts on this. 
please consider opening a question on: https://electronics.stackexchange.com
